MapboxNavigation/SpeedLimit.swift

```diff
@@ -0,0 +1,161 @@
+                .font: UIFont.boldSystemFont(ofSize: fontSize * 0.7)
+                ])
+            labels.append(unitLabel)
+            
+            updateStackView(with: labels)
+        }
+    }
+    
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        commonInit()
+    }
+    
+    required public init(coder: NSCoder) {
+        super.init(coder: coder)
+        commonInit()
+    }
+    
+    func commonInit() {
+        isLayoutMarginsRelativeArrangement = true
+        axis = .vertical
+        alignment = .center
+        contentMode = .scaleAspectFill
+        distribution = .fillProportionally
+    }
+    
+    func updateStackView(with labels: [SpeedLimitLabel]) {
+        subviews.forEach { $0.removeFromSuperview() }
+        addArrangedSubviews(labels)
+        layoutIfNeeded()
+    }
+    
+    func addBackground() {
+        let background: UIView
+        if region == .unitedStates {
+            background = UnitedStatesSignBase(frame: bounds)
+        } else {
+            background = WorldSignBase(frame: circleBounds)
+        }
+        insertSubview(background, at: 0)
+    }
+
+    open override func layoutSubviews() {
+        super.layoutSubviews()
+        addBackground()
+    }
+}
+
+public enum SpeedLimitSignRegionType {
+    case world
+    case unitedStates
+}
+
+class WorldSignBase: UIView {
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        commonInit()
+    }
+    
+    func commonInit() {
+        backgroundColor = .white
+        layer.cornerRadius = bounds.size.width / 2
+        layer.borderWidth = 5
+        layer.borderColor = UIColor(red:0.93, green:0.11, blue:0.14, alpha:1.0).cgColor
+        autoresizingMask = [.flexibleWidth, .flexibleHeight]
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        commonInit()
+    }
+}
+
+
+class UnitedStatesSignBase: UIView {
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        commonInit()
+    }
+    
+    func commonInit() {
+        backgroundColor = .white
+        layer.cornerRadius = 3
+        layer.borderWidth = 1
+        layer.borderColor = UIColor.black.cgColor
+        autoresizingMask = [.flexibleWidth, .flexibleHeight]
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        commonInit()
+    }
+}
+
+extension SpeedUnit {
+    var localizedSpeedUnit: String {
+        switch self {
+        case .kilometersPerHour:
+            return NSLocalizedString("KILOMETERS_PER_HOUR", bundle: .mapboxNavigation, value: "kph", comment: "Inform the user on limits for speed.")
+        case .milesPerHour:
+            return NSLocalizedString("MILES_PER_HOUR", bundle: .mapboxNavigation, value: "mph", comment: "Inform the user on limits for speed.")
```

Comment:
@bsudekum perhaps the comments should be unique to each localized string? `Inform the user on limits for speed` should probably indicate one is `mph` or `kph` 💭